### PR TITLE
Add reset Docker storage per BZ-1501504

### PR DESCRIPTION
**Pull request by mburke5678**
WIP first draft
https://bugzilla.redhat.com/show_bug.cgi?id=1501504

### CODE REVIEW

**Comment by mburke5678** [comment]
@jupierce PTAL

**Comment by mburke5678** [comment]
@mdshuai 
Can you take a look at the PR as you were the QA contact on the original PR? | Collapse All

File: admin_guide/manage_nodes.adoc [hunk 1]
```diff
@@ -487,6 +487,82 @@ default value for `max-pods` is `250`. This means that unless the node has 25
...
+. Run the following command to shut down Docker and the *atomic-openshift-node* service:
++
+----
+$ systemctl stop docker atomic-openshift-node
+----
+
+. Run the following command to shut down Docker and the the *atomic-openshift-node* service:
++
+----
+$ systemctl stop docker atomic-openshift-node
```
Comment (mdshuai) [inline admin_guide/manage_nodes.adoc]: Total same step with above?

**Comment by mdshuai** [comment]
I think you can use the step @jupierce provided.  One thing need update is you can use `oadm cordon/uncordon` to make node as `unschedulable/schedulable`

**Comment by mburke5678** [comment]
@mdshuai We don't refer to oadm cordon in the docs. In the same topic, we use `oadm manage-node <node> --schedulable=false/true` to mark a node as unschedulable.  Is that the correct command?

**Comment by mburke5678** [comment]
@jupierce You code has a line to remove the volume directory:
`rm -rf /var/lib/origin/openshift.local.volumes`

Should we include that?

**Comment by jupierce** [comment]
@sjenning We deleted local volumes during our interactive work on the nodes. Do you want to include it in the docs?

**Comment by sjenning** [comment]
@jupierce yes, I think so.  If we are going to document this, might as well include everything need to get the node back to a pristine storage condition.

**Comment by mburke5678** [comment]
@jupierce @sjenning Is there any negative consequence to deleting the local volumes? Can this be an optional step, if the user wants to retain them? Does stopping Docker make these volumes unneeded as any associated containers would be wiped?

**Comment by sjenning** [comment]
@mburke5678 it clears the local image cache so images, including infra ose-* images, will need to be repulled and might result in slower pod start times while the image store warms back up.

I would consider this to be an "all or nothing" procedure.  Picking a choosing things to delete might result in weird state.  I don't know all the assumptions docker makes.

**Comment by mburke5678** [comment]
@jupierce @mdshuai @sjenning Can these changes roll back to OCP 3.4? 
I am not sure when docker-storage-setup was added.

**Comment by mburke5678** [comment]
[rev_history]
|xref:../admin_guide/manage_nodes.adoc#admin-guide-manage-nodes[Manging Nodes]
|Added new section on xref:../admin_guide/manage_nodes.adoc#managing-nodes-docker-reset[Resetting Docker Storage] to free up space on nodes.
%

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

File: admin_guide/manage_nodes.adoc [hunk 2]
```diff
@@ -487,6 +487,90 @@ default value for `max-pods` is `250`. This means that unless the node has 25
...
+----
+
+One solution to this problem is to reset Docker storage to remove artifacts not needed by Docker.
+
+On the node where you want to restart Docker storage:
+
+. Run the following command to mark the node as unschedulable:
++
+----
+oadm manage-node <node> --schedulable=false
```
Comment (bmcelvee) [inline admin_guide/manage_nodes.adoc]: $

File: admin_guide/manage_nodes.adoc [hunk 3]
```diff
@@ -487,6 +487,90 @@ default value for `max-pods` is `250`. This means that unless the node has 25
...
+. Run the following command to shut down Docker and the *atomic-openshift-node* service:
++
+----
+$ systemctl stop docker atomic-openshift-node
+----
+
+. Run the following command to remove the local volume directory:
++
+----
+rm -rf /var/lib/origin/openshift.local.volumes
```
Comment (bmcelvee) [inline admin_guide/manage_nodes.adoc]: $

**Comment by bmcelvee** [comment]
@mburke5678 The example log is building strangely for me, entirely red text with red highlighting. Is it building that way for you?

Otherwise LGTM! 🍂

**Comment by mburke5678** [comment]
@bmcelvee Thanks. I changed a [source,yaml] to [source,bash] for those example commands. Does that look better?

**Comment by bmcelvee** [comment]
@mburke5678 Yes, better! 😸

**Comment by vikram-redhat** [comment]
@mburke5678 - you had enterprise 3.4 as one of the branches where this change applies, but you haven't picked it to that branch. Also, does it still need the peer review needed label, even though it has been merged?

**Comment by mburke5678** [comment]
@vikram-redhat Done. Thanks.

**Pull request merged**